src/command_modules/azure-cli-acr/azure/cli/command_modules/acr/build.py

```diff
@@ -47,6 +47,9 @@ def acr_build(cmd,  # pylint: disable=too-many-locals
     from ._client_factory import cf_acr_registries
     client_registries = cf_acr_registries(cmd.cli_ctx)
 
+    if os_type and platform:
+        raise CLIError("[--os] has been depricated. Please use [--platform] instead.")
```

Comment:
There is a dedicated mechanism in the CLI for this situation:
https://github.com/Azure/azure-cli/blob/dev/doc/authoring_command_modules/authoring_commands.md#deprecating-commands-and-arguments

The previous logic was "both could be used but had to match".  The current logic is "--os can't be used". This is a breaking change. A more logical way would be for `--os` and `--platform` to reference the same variable, but have `--os` deprecated in favor of `--platform`.  It would look something like:

```Python
with self.argument_context(scope) as c:
  c.argument('platform', options_list=['--platform', c.deprecate(target='--os', redirect='--platform', hide=True)])
```

In this way, --platform no longer appears, but does work, and fills the same value as using `--os` would.

Reply:
So as opposed to introducing a separate argument `--platform` alongside `--os`, we should remove the `--os` parameter and add it to the options of `--platform` with the deprecate wrapper? What do you mean by "--platform no longer appears"? Do you mean --os no longer appears, but does work, and fills the same value as using --platform would?

Reply:
If you use `hide=True` then `--os` won't appear when using `-h` but it will still work. It is effectively an alias to `--platform` and the user will get a warning if they use it that it is deprecated and they should use `--platform` instead.

Reply:
If your situation is different feel free to not use it, but it seems like it fits your scenario.

Reply:
I understand. This is definitely a better option for maintaining the code as well. Will make this change of integrating `--os` into `--platform`. //cc @shahzzzam